src/Mod/PartDesign/App/FeatureSketchBased.cpp

```diff
@@ -1012,7 +1012,7 @@ double ProfileBased::getReversedAngle(const Base::Vector3d &b, const Base::Vecto
 }
 
 void ProfileBased::getAxis(const App::DocumentObject *pcReferenceAxis, const std::vector<std::string> &subReferenceAxis,
-                          Base::Vector3d& base, Base::Vector3d& dir)
+                          Base::Vector3d& base, Base::Vector3d& dir, bool checkPerpendicular)

```

Comment:
For the case of a helix, it is not wrong to have helix axis perpendicular to the sketch. Hence the need for this option. Defaults to true, to maintain api compatibility with old code.